pkg/scheduler/apis/config/validation/validation.go

```diff
@@ -97,10 +97,50 @@ var removedPluginsByVersion = []removedPlugins{
 	},
 	{
 		schemeGroupVersion: v1beta2.SchemeGroupVersion.String(),
-		plugins:            []string{"NodeLabel", "ServiceAffinity", "NodePreferAvoidPods"},
+		plugins: []string{
+			"NodeLabel",
+			"ServiceAffinity",
+			"NodePreferAvoidPods",
+			"NodeResourcesLeastAllocated",
+			"NodeResourcesMostAllocated",
+			"RequestedToCapacityRatio",
+		},
 	},
 }
 
+// conflictPluginsByVersion maintains a map of conflict plugins in each version.
+// Remember to add an entry to that list when creating a new component config
+// version (even if the list of conflict plugins is empty).
+var conflictPluginsByVersion = map[string]map[string]sets.String{
+	v1beta1.SchemeGroupVersion.String(): {
+		"NodeResourcesFit": sets.NewString(
```

Comment:
Today, in v1beta1, people can use NodeResourcesFit and the other plugins. We should continue allowing that.

Reply:
Just deprecating the plugins in v1beta2 is enough.

Reply:
I don't think so, we should prevent using both in v1beta1.

Reply:
ah, we should only do that for score indeed!

Reply:
lets just call this function for "profile.Plugins.Score" and rename the function as such: `validateScorePluginSetForConflictPlugins`

Reply:
back-dating plugin conflicts into already-released scheduler configs is _really_ hard to reason about, and seems likely to break a v1beta1 config file that worked in 1.21 and which we still claim to support in 1.22

if the scoring capability being added to NodeResourcesFit is taking the place of those three separate plugins, it seems more straightforward to:
1. omit the `scoringStrategy` field from the NodeResourcesFit v1beta1 params
2. make NodeResourcesFit keep 1.21 behavior if the `scoringStrategy` field is nil
3. disallow the three individual plugins in v1beta2+

Reply:
Ideally we want `ScoringStrategy` in the internal type to not be a reference, which I am planning to change it in a followup PR. Assuming it is not a reference, if a user lists NodeResourcesFit as a scoring plugin, then we will get all sorts of surprising behavior because if it gets properly defaulted, then the pod will get scored by both the legacy plugin (e.g., LeastAllcoated) and the new one (NodeResourcesFit), which is a configuration we don't want users to have.

Reply:
I want to stress that this conflict check is backward compatible because NodeResourcesFit didn't exist as a scoring plugin in the released versions.

Reply:
> make NodeResourcesFit keep 1.21 behavior if the scoringStrategy field is nil

My concern is that if the user specifies NodeResourcesFit in the Score PluginSet in a v1beta1 configuration, they would be surprised if that resulted in a no-op since NodeResourcesFit now implements the score extension point. I get that this is the existing v1beta1 behavior, but I bet it still be surprising.

Reply:
An alternative is to disallow the use of NodeResroucesFit as a score plugin completely in v1beta1. Which sadly introduces new validation machinery as well.
OTOH, to learn about the new capabilities of NodeResourcesFit, users have to get it from the documentation, where we can stress the fact that in v1beta1 is no-op and they have to migrate to v1beta2. I think this is reasonable.
But ultimately, I'm not heavily opposed to adding the conflict detection. The problem is that it just sets a precedent in the k8s API that we don't want to encourage.

Reply:
> But ultimately, I'm not heavily opposed to adding the conflict detection. The problem is that it just sets a precedent in the k8s API that we don't want to encourage.

The good thing is that this is not permanent, we will remove it with the v1beta1 deprecation.